graph/internal/set/same.go

```diff
@@ -14,24 +14,15 @@ import "unsafe"
 // hash maps are passed as a pointer to a runtime Hmap struct. A map is
 // not seen by the runtime as a pointer though, so we use unsafe to get
 // the maps' pointer values to compare.
-func same(a, b Nodes) bool {
+func same[T comparable](a, b Set[T]) bool {
```

Comment:
If `graph.Node` and the integer sets can't go into the same set of functions without unreasonable cost, I'd prefer that we just retained the integer and node set types, the integer being type parameterised on `interface{ int | int64}` rather than on `comparable`. I don't see the need for full comparable support and the more conservative approach means that the names in this package stay the same for the most part.

The package doc comment change would be reverted in this case.

Reply:
@kortschak Okay, that makes sense.  I remember there being an implementation reason why I couldn't merge all set types together, but I'm not sure anymore, so I'll have a look again soon.

Reply:
Yeah, it's because a normal set is a `map[int|int64]struct{}` whereas a node set is a `map[int64]graph.Node` (in the former, set values are map keys; in the latter, set values are map values indexed by node IDs), so there's a mismatch and I don't know how to de-duplicate things further.

So I've changed `T` to `interface{ int | int64 }` and renamed some things back as requested.

Reply:
This can be done by type switching. Until this can be done without `any(v).(type)`, it is not something that we want to do. So leaving it at this is fine for now.